datacube/utils/geometry/_base.py

```diff
@@ -213,11 +216,16 @@ def to_epsg(self) -> Optional[int]:
         """
         EPSG Code of the CRS or None
         """
+        if self._epsg is not False:
+            return self._epsg
+        self._epsg = self._crs.to_epsg()
+        if self._epsg is None:
+            self._epsg = False
         return self._epsg
```

Comment:
If None is a valid return value, why set to False?  Shouldn't we just cache and return it as normal?

I think L222-223 need to go.